File: apps/services/flow/backend/src/Flow.Domain/Entities/Notification.cs
using Flow.Domain.Common;

namespace Flow.Domain.Entities;

public class Notification : BaseEntity
{
    public Guid? TaskId { get; set; }
    public Guid? WorkflowDefinitionId { get; set; }
    public string Type { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public string? TargetUserId { get; set; }
    public string? TargetRoleKey { get; set; }
    public string? TargetOrgId { get; set; }
    public string Status { get; set; } = NotificationStatus.Unread;
    public string SourceType { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ReadAt { get; set; }

    public TaskItem? Task { get; set; }
    public FlowDefinition? WorkflowDefinition { get; set; }
}

public static class NotificationStatus
{
    public const string Unread = "Unread";
    public const string Read = "Read";
}

public static class NotificationType
{
    public const string TaskAssigned = "TASK_ASSIGNED";
    public const string TaskReassigned = "TASK_REASSIGNED";
    public const string TaskTransitioned = "TASK_TRANSITIONED";
    public const string AutomationSucceeded = "AUTOMATION_SUCCEEDED";
    public const string AutomationFailed = "AUTOMATION_FAILED";
    public const string WorkflowAssigned = "WORKFLOW_ASSIGNED";
}

public static class NotificationSourceType
{
    public const string WorkflowTransition = "WorkflowTransition";
    public const string AutomationHook = "AutomationHook";
    public const string Assignment = "Assignment";
    public const string System = "System";
}

File: apps/services/flow/frontend/src/types/notification.ts
export const NOTIFICATION_TYPES = [
  "TASK_ASSIGNED",
  "TASK_REASSIGNED",
  "TASK_TRANSITIONED",
  "AUTOMATION_SUCCEEDED",
  "AUTOMATION_FAILED",
  "WORKFLOW_ASSIGNED",
] as const;
export type NotificationType = typeof NOTIFICATION_TYPES[number];

export const NOTIFICATION_TYPE_LABELS: Record<NotificationType, string> = {
  TASK_ASSIGNED: "Assigned",
  TASK_REASSIGNED: "Reassigned",
  TASK_TRANSITIONED: "Transitioned",
  AUTOMATION_SUCCEEDED: "Automation OK",
  AUTOMATION_FAILED: "Automation Failed",
  WORKFLOW_ASSIGNED: "Workflow Assigned",
};

export const NOTIFICATION_TYPE_COLORS: Record<NotificationType, string> = {
  TASK_ASSIGNED: "bg-blue-100 text-blue-700",
  TASK_REASSIGNED: "bg-indigo-100 text-indigo-700",
  TASK_TRANSITIONED: "bg-green-100 text-green-700",
  AUTOMATION_SUCCEEDED: "bg-emerald-100 text-emerald-700",
  AUTOMATION_FAILED: "bg-red-100 text-red-700",
  WORKFLOW_ASSIGNED: "bg-purple-100 text-purple-700",
};

export interface NotificationResponse {
  id: string;
  taskId?: string | null;
  workflowDefinitionId?: string | null;
  type: NotificationType;
  title: string;
  message: string;
  targetUserId?: string | null;
  targetRoleKey?: string | null;
  targetOrgId?: string | null;
  status: "Unread" | "Read";
  sourceType: string;
  createdAt: string;
  readAt?: string | null;
}

export interface NotificationPagedResponse {
  items: NotificationResponse[];
  totalCount: number;
  page: number;
  pageSize: number;
}

export interface NotificationSummary {
  unreadCount: number;
}

export const NOTIFICATION_SOURCE_TYPES = [
  "Assignment",
  "WorkflowTransition",
  "AutomationHook",
  "System",
] as const;
export type NotificationSourceType = typeof NOTIFICATION_SOURCE_TYPES[number];
